umbraco/Umbraco-CMS
# Migrates preview auth Middleware

Migrates UriExtensionsTests to netcore, fixes preview controller bits, adds tests for preview path for back office route check, fixes virtual paths for views.

## Testing

* Ensure all tests pass
* Start the back office, create or go to an existing document
* Add a breakpoint to PreviewAuthenticationMiddleware.InvokeAsync beneath the `if (isPreview)` check
* Click save & preview
* Your breakpoint will be hit because:
  * This is actually a front-end request
  * There's a preview cookie set
* Step through the code and ensure that `context.User.AddIdentity(backOfficeIdentity);` gets executed
